src/eca/features/tools.clj

```diff
@@ -19,38 +19,75 @@
 
 (def ^:private logger-tag "[TOOLS]")
 
+;; ============================================================================
+;; Tool Registry & Configuration
+;; ============================================================================
+
+(def ^:private native-tool-registry
+  [{:key :filesystem :definitions f.tools.filesystem/definitions}
+   {:key :shell      :definitions f.tools.shell/definitions}
+   {:key :editor     :definitions f.tools.editor/definitions}])
+
+;; ============================================================================
+;; Native Tool Loading & Processing
+;; ============================================================================
+
+(defn ^:private load-user-tools [config]
+  (some-> (find-ns 'eca.features.tools.user)
+          (ns-resolve 'user-tool-definitions)
+          (apply [config])))
+
+(defn ^:private load-enabled-native-tools [config]
+  (->> native-tool-registry
+       (filter #(get-in config [:nativeTools (:key %) :enabled]))
+       (map :definitions)
+       (apply merge {})))
+
+(defn ^:private substitute-workspace-roots [db description]
+  (string/replace description #"\\$workspaceRoots"
+                  (constantly (tools.util/workspace-roots-strs db))))
+
+(defn ^:private transform-tool-definition [db [name tool]]
+  [name (-> tool
+            (assoc :name name)
+            (update :description (partial substitute-workspace-roots db)))])
+
 (defn ^:private native-definitions [db config]
```

Comment:
we should keep this function as it is, and add a new one for the custom tools in [this concat](https://github.com/editor-code-assistant/eca/pull/102/files#diff-60301e9fa63d939355dc6a869677d9d75758758773918ba192f17888c9aef79cR91)